src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/whenshouldbeused/InputWhenShouldBeUsedSwitchExpression.java

```diff
@@ -0,0 +1,108 @@
+/*
+WhenShouldBeUsed
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.whenshouldbeused;
+
+public class InputWhenShouldBeUsedSwitchExpression {
+
+    void test(Object j, int x) {
+        int a = switch (j) {
+            case Point(int _, int y): // violation, 'When should be used instead of a single if*.'
+                if (y == 0) {
+                    yield 0;
+                }
+            case Integer _: if (x == 0)  yield 4;
+            // violation above, 'When should be used instead of a single if*.'
+            default: { yield 3;}
+        };
+        int b = switch (j) {
+            case Point(int _, int y): {// violation, 'When should be used instead of a single if*.'
+                if (y == 0) {
+                    yield 0;
+                }
+            }
+            // violation below, 'When should be used instead of a single if*.'
+            case ColoredPoint(Point(_, _), String _) : {
+                if (x == 0) yield 4;
+            }
+            default: { yield 3;}
+        };
+        int bb = switch (j) {
+            case Point(int _, int y) when y == 0:  {
```

Comment:
Add or point to an example where a `when` has a single if statement inside. Will we get a violation?

Reply:
> Will we get a violation?

Yes because you can combine the if condition with when. I added an example for this

Reply:
https://github.com/checkstyle/checkstyle/blob/d2fb8bada0ea4998b6643a4660b676f76d1c543e/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/whenshouldbeused/InputWhenShouldBeUsedSwitchRule.java#L98-L111

Reply:
Since we violate this case, to me it seems to me that the message is slightly misleading. We say use when instead of if but they are already using when. I am not sure if we should change it, but I am pointing it out.